initrd/bin/network-init-recovery

```diff
@@ -28,34 +45,37 @@ if [ -e /sys/class/net/eth0 ]; then
 	# Set up static IP
 	if [ ! -z "$CONFIG_BOOT_STATIC_IP" ]; then
 		ifconfig eth0 $CONFIG_BOOT_STATIC_IP
-	#Get ip from DHCP
-	elif [ -e /sbin/udhcpc ];then
-		if udhcpc -T 1 -q; then
-			if [ -e /sbin/ntpd ]; then
-				DNS_SERVER=$(grep nameserver /etc/resolv.conf|awk -F " " {'print $2'})
-				killall ntpd 2&>1 > /dev/null
-		 		if ! ntpd -d -N -n -q -p $DNS_SERVER > /dev/ttyprintk; then
-					if ! ntpd -d -d -N -n -q -p ntp.pool.org> /dev/ttyprintk; then
-						echo "NTP sync unsuccessful." > /dev/tty0
-					fi
+	fi
+	dev=eth0
+fi
+
+#Get ip from DHCP
+if [ -e /sbin/udhcpc ]; then
+	if udhcpc -T 1 -i $dev -q; then
+		if [ -e /sbin/ntpd ]; then
```

Comment:
Did not test myself, but it looks like if you have `CONFIG_BOOT_STATIC_IP` set, it'll now incorrectly try to invoke udhcpc.

Maybe you should move the CONFIG_BOOT_STATIC_IP test down with the DHCP logic, so CONFIG_BOOT_STATIC_IP could apply to any network device (not just eth0) and it'd know not to invoke udhcpc.

Reply:
@JonathonHall-Purism I have move CONFIG_BOOT_STATIC_IP down as requested.